### PR TITLE
Feature 1699: Add support for views

### PR DESCRIPTION
**Pull request by spasepremcheski**
## Summary

Adding support for Views. The main idea is to be able to add screenshots of the views annotated with @ArkiveView inside the Arkive Components showcase.

### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

This PR:
- Adds @ArkiveView annotation together with ArkiveViewCollector
- Adding tag for composable or view during the collecting of the annotated functions
- Generate code for running view tests with UiComponentSpec
- Add ViewCollector to Subprocessor and viewTestFunction to ArkiveTestProcessor
- Create ComponentRepository
- Move MetaDataSpec call to separate Subprocessor
- Add example for annotating a view with @ArkiveView (show_card_view)

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

#### `skipPreviews` set to false
Screenshot 2025-05-06 at 09 00 18

#### `skipPreviews` set to false, but `RoundedButton` previews have @ArkiveComposable removed
Screenshot 2025-05-06 at 09 03 30

#### `skipPreviews` set to true, but `RoundedButton` previews have @ArkiveComposable removed
**There seems to be a bug here connected to the `skipPreviews` argument**
Screenshot 2025-05-06 at 09 02 55

### CODE REVIEW

File: processor/src/main/java/com/infinum/arkive/processor/ArkiveProcessor.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ class ArkiveProcessor(
         }
         processed = true
 
-        ComposeSubprocessor().process(resolver, codeGenerator, logger, options)
+        UiComponentSubprocessor().process(resolver, codeGenerator, logger, options)
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/ArkiveProcessor.kt]: I would go with two processors instead of combining both

ComposeSubProcessor
ViewSubProcessor 

This goes with everything else. Like holders, there's no need to have one big holder. This might go wrong in the future, so it's better to separate them 

ComposableHolder
ViewHolder 
or something similar

But then, if you needed to add both of them to the showcase, then extract the info from each one of them

File: processor/src/main/java/com/infinum/arkive/processor/specs/UiComponentSpec.kt [hunk 2]
```diff
@@ -2,28 +2,32 @@ package com.infinum.arkive.processor.specs
...
 import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.INT
 import com.squareup.kotlinpoet.LambdaTypeName
 import com.squareup.kotlinpoet.MemberName
 import com.squareup.kotlinpoet.TypeSpec
 import com.squareup.kotlinpoet.UNIT
 import com.squareup.kotlinpoet.ksp.writeTo
 
-class ComposeSpec(
+class UiComponentSpec(
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/specs/UiComponentSpec.kt]: I would aslo go with two different specs here, this viliats the SRP 

this class now has two different reasons to change compose and view :D 
so let's separate them

File: processor/src/main/java/com/infinum/arkive/processor/specs/UiComponentSpec.kt [hunk 3]
```diff
@@ -65,7 +69,27 @@ class ComposeSpec(
...
+                    parameters = arrayOf(
+                        ClassName("kotlin", "String"),
+                        LambdaTypeName.get(returnType = INT),
+                    ),
+                    returnType = UNIT,
+                ),
+            )
+            .addCode(
+                holders.filter { it.tags.contains(TAG_VIEW) }.joinToString(separator = "\n") { holder ->
                     getRunnerFunction(holder)
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/specs/UiComponentSpec.kt]: Move to the new spec ViewSpec

File: processor/src/main/java/com/infinum/arkive/processor/subprocessors/UiComponentSubprocessor.kt [hunk 4]
```diff
@@ -4,33 +4,36 @@ import com.google.devtools.ksp.processing.CodeGenerator
...
                 addAll(previewCollector.collect())
             }
         }
 
-        with(validator.validate(composeHolders)) {
-            ComposeSpec(codeGenerator, this, logger).write()
-            ComposeMetaDataSpec(codeGenerator, this).write()
+        with(validator.validate(uiComponentHolders)) {
+            UiComponentSpec(codeGenerator, this, logger).write()
+            UiComponentMetaDataSpec(codeGenerator, this).write()
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/subprocessors/UiComponentSubprocessor.kt]: I'm completely okay with MetaData having one single spec, as it's generating the same file. However, everything else should be separated, even the sub-processor 

Let's set a rule here.
If you're collecting different annotations and generating different files, then use a different sub-processor

Then, for `UiComponentMetaDataSpec' you can have a new subprocessor for MetaData that collects both and generates a single metadata file.

**Review by thisAAY — CHANGES_REQUESTED** [review]
Great work 🥇 
The functionality is completely correct, we just need to set some separation for future proofing

**Comment by spasepremcheski** [comment]
I separated the Subprocessors, Spec and Holder to separate View and Compose logic. 

After a quick sync with Ahmed the following things were resolved:
There was a drawback to this, that the `MetaDataSpec` couldn't be called twice and create the same file twice. To overcome this the `MetaDataSubprocessor` was created and the `ViewHolder` and `ComposeHolder` inherit from a `Holder` interface that is then passed to the `MetaDataSubprocessor`. In order to not do collection of Views and Composables twice, the `ComponentRepository` object is created. At the end of the `ArkiveProcessor.process()` this `ComponentRepository` object's data is cleared. The functionality remains the same, but the implementation is more clear

File: processor/src/main/java/com/infinum/arkive/processor/specs/ComposeSpec.kt [hunk 5]
```diff
@@ -65,7 +66,7 @@ class ComposeSpec(
                 ),
             )
             .addCode(
-                holders.joinToString(separator = "\n") { holder ->
+                holders.filter { it.tags.contains(TAG_COMPOSABLE) }.joinToString(separator = "\n") { holder ->
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/specs/ComposeSpec.kt]: Why filtering? we already getting only a set of Compose holders

File: processor/src/main/java/com/infinum/arkive/processor/specs/ViewSpec.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                LambdaTypeName.get(
+                    parameters = arrayOf(
+                        ClassName("kotlin", "String"),
+                        LambdaTypeName.get(returnType = INT),
+                    ),
+                    returnType = UNIT,
+                ),
+            )
+            .addCode(
+                holders.filter { it.tags.contains(TAG_VIEW) }.joinToString(separator = "\n") { holder ->
```
Comment (thisAAY) [inline processor/src/main/java/com/infinum/arkive/processor/specs/ViewSpec.kt]: Same here

**Review by thisAAY — APPROVED** [review]
I approved it, just one more comment that instead of having the Repo as object, we can create it the main processor and pass it to each sub  processor

Instead of having an object pattern

File: processor/src/main/java/com/infinum/arkive/processor/specs/ComposeSpec.kt (hunk 5 above)
Comment (spasepremcheski) [reply processor/src/main/java/com/infinum/arkive/processor/specs/ComposeSpec.kt]: I probably forgot to remove it from the previous implementation.
